targets/TARGET_NXP/TARGET_MCUXpresso_MCUS/TARGET_LPC/flash_api.c

```diff
@@ -0,0 +1,127 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "flash_api.h"
+#include "mbed_critical.h"
+
+#if DEVICE_FLASH
+
+#include "fsl_flashiap.h"
+
+int32_t flash_init(flash_t *obj)
+{
+    return 0;
+}
+
+int32_t flash_free(flash_t *obj)
+{
+    return 0;
+}
+
+int32_t flash_erase_sector(flash_t *obj, uint32_t address)
+{
+    uint32_t n;
+    uint32_t status;
+    int32_t ret = -1;
+
+    /* We need to prevent flash accesses during erase operation */
+    core_util_critical_section_enter();
+
+    n = address / FSL_FEATURE_SYSCON_FLASH_SECTOR_SIZE_BYTES;   // Get Sector Number
+
+    status = FLASHIAP_PrepareSectorForWrite(n, n);
+    if (status == kStatus_FLASHIAP_Success) {
+        status = FLASHIAP_EraseSector(n, n, SystemCoreClock);
+        if (status == kStatus_FLASHIAP_Success) {
+            ret = 0;
+        }
+    }
+
+    core_util_critical_section_exit();
+
+    return ret;
+}
+
+int32_t flash_program_page(flash_t *obj, uint32_t address, const uint8_t *data, uint32_t size)
+{
+    uint32_t n;
+    uint32_t status;
+    int32_t ret = -1;
+    uint8_t buf[FSL_FEATURE_SYSCON_FLASH_PAGE_SIZE_BYTES];
+
+    if (address == 0) {                              // Check for Vector Table
```

Comment:
How does this work with bootloader, or does not have any affect? (not sure what `n` represents)

From the code, `n` gets overwritten on line 81 (is it just reusing the variable? wouldn't it be more clear to have two separate ? `n` on the line 81 represents sector number, here on this line it is signature calculation?

Reply:
> You are right, the variable is reused for a different purpose. If it is not clear we can create a different variable.

Please do, should help clarity in this function. The rest LGTM